discover/infer.go

```diff
@@ -0,0 +1,77 @@
+package discover
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	effx_api "github.com/effxhq/effx-api-v2-go/client"
+	"github.com/effxhq/effx-cli/data"
+	"github.com/thoas/go-funk"
+)
+
+var serviceDirectoryNames = []string{
+	"services",
+	"apps",
+}
+
+var relavantFiles = []string{
+	"go.mod",           // golang/gin
+	"package.json",     // nodejs
+	"requirements.txt", // python/django/flask
+	"pom.xml",          // java/maven
+	"gemfile",          // rails, sinatra
+	"composer.json",    // larvel
+	"build.gradle",     // java spring
+	"mix.exs",          // elixer/phoenix
+}
+
+// DetectServicesFromRelavantFiles detects services based on
+// containing a service-like file (package.json etc)
+func DetectServicesFromRelavantFiles(workdir string, effxFiles []data.EffxYaml, sourceName string) ([]effx_api.DetectedServicesPayload, error) {
+	detectedServices := []effx_api.DetectedServicesPayload{}
+	effxFileLocations := filePathsFromEffxYaml(effxFiles)
+
+	err := filepath.Walk(workdir, func(path string, f os.FileInfo, err error) error {
+		if f.IsDir() {
+			dirName := f.Name()
+
+			if directoryContainsEffxYaml(f, effxFileLocations) {
+				return nil
+			}
+
+			// if directory name contain "services", "apps" etc, all subdirectories
```

Comment:
It would be nice if this was configurable through environment variables or a config file. Each project can then declare its intended structure.